src/components/app.js

```diff
@@ -1,36 +1,173 @@
-import icon from '../assets/menu.png';
-
 const addList = document.querySelector('.list-items');
+const btn = document.querySelector('.add-todo');
+const addElem = document.querySelector('.add-list');
+
+let lists;
+
+const saveItems = () => {
+  localStorage.setItem('data', JSON.stringify(lists));
+};
+
+const savedItems = JSON.parse(localStorage.getItem('data'));
+
+if (Array.isArray(savedItems)) {
+  lists = savedItems;
+} else {
+  lists = [];
+}
+
+class Node {
+  constructor(description) {
+    this.description = description;
+    this.completed = false;
+    this.index = `${lists.length}`;
+  }
+}
+
+const createItems = (description) => {
+  const newList = new Node(description);
+  lists.push(newList);
+
+  saveItems();
+};
 
-const todos = [
-  {
-    description: 'wash dishes',
-    completed: false,
-    index: '1',
-  }, {
-    description: 'complete To Do list project',
-    completed: false,
-    index: '2',
-  },
-];
+const setEditing = (itemId) => {
+  lists.forEach((list) => {
+    if (list.index === itemId) list.isEditing = true;
+  });
+
+  saveItems();
+};
+
+const updateItem = (itemId, newdescription) => {
+  lists.forEach((list) => {
+    if (list.index === itemId) {
+      list.description = newdescription;
+      list.completed = false;
+      list.length = `${lists.length}`;
+      list.isEditing = false;
+    }
+  });
+  saveItems();
+};
+
+const deleteItems = (listId) => {
+  lists = lists.filter((list) => list.index !== listId);
+  saveItems();
+};
```

Comment:
- [ ] Kindly create a new JavaScript file for adding, deleting and editing functionalities as requested in the project description:
![image](https://user-images.githubusercontent.com/91486791/220932623-cff891fa-a0e7-4834-baed-df8de4cd046f.png)